webapp/packages/plugin-codemirror6/src/ReactCodemirror.tsx

```diff
@@ -136,14 +138,27 @@ export const ReactCodemirror = observer<IReactCodeMirrorProps, IEditorRef>(
     });
 
     useLayoutEffect(() => {
-      if (value !== undefined && view && value !== view.state.doc.toString()) {
+      if (view && value !== undefined && value !== view.state.doc.toString()) {
         view.dispatch({
           changes: { from: 0, to: view.state.doc.length, insert: value },
           annotations: [External.of(true)],
         });
       }
     }, [value, view]);
 
+    useLayoutEffect(() => {
+      if (
+        view &&
+        memoizedCursor &&
+        (view.state.selection.main.anchor !== memoizedCursor.anchor || view.state.selection.main.head !== memoizedCursor.head)
+      ) {
+        view.dispatch({
+          selection: memoizedCursor,
+          annotations: [External.of(true)],
+        });
+      }
+    }, [view, memoizedCursor]);
```

Comment:
it's better to join this two because we want to be sure that they will be executed together you can conditionally set selection and/or changes together